block/data_cache.go

```diff
@@ -0,0 +1,65 @@
+package block
+
+import (
+	"sync"
+
+	"github.com/rollkit/rollkit/types"
+)
+
+// DataCache maintains data that are seen and hard confirmed
+type DataCache struct {
+	data       *sync.Map
+	hashes     *sync.Map
+	daIncluded *sync.Map
+}
+
+// NewDataCache returns a new DataCache struct
+func NewDataCache() *DataCache {
+	return &DataCache{
+		data:       new(sync.Map),
+		hashes:     new(sync.Map),
+		daIncluded: new(sync.Map),
+	}
+}
+
+func (hc *DataCache) getData(height uint64) (*types.Data, bool) {
+	data, ok := hc.data.Load(height)
+	if !ok {
+		return nil, false
+	}
+	return data.(*types.Data), true
+}
+
+func (hc *DataCache) setData(height uint64, data *types.Data) {
+	if data != nil {
+		hc.data.Store(height, data)
+	}
+}
```

Comment:
put another way, I think this `nil` check is a hidden assumption of what should be stored in the map.

if it was meant as input validation, then it should return an error, and not be handled silently.

Reply:
i still think its better to keep the check, mainly because at the callsite (trySyncNextBlock) we only check the status of `getData` and not the actual data being `nil`. This could lead to `nil` pointer deref. so, this check would be a additional measure, although in no circumstances, we should try to setData which is `nil`.

Reply:
i think its better to remove the status itself and only rely on the data then we can check for the data being `nil` and refactor the set methods to set it even if its nil.